### PR TITLE
Create application to catalog things

### PR DESCRIPTION
**Pull request by bengigo**
**Changes made in the application** 

_Group task_
- Create Item class in a separate .rb file.
- All Item class properties visible in the diagram should be defined and set up in the constructor method. - Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. They should have a custom setter method created.
- Implement methods:
   - can_be_archived?() in the Item class
   - should return true if published_date is older than 10 years, otherwise, it should return false
   - move_to_archive() in the Item class
   - should reuse can_be_archived?() method
   - should change the archived property to true if the result of the can_be_archived?() method is true, should do nothing if the result of the can_be_archived?() method is false
- Create a main.rb file that will serve as your console app entry-point.
- Implement startup actions:
   - Present the user with a list of options to perform.
   - Let users choose an option.
   - If needed, ask for parameters for the option.
   - Have a way to quit the app.
   
   
_@Ney2_ 
- Create a Book class in a separate .rb file.
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
   - add_item method in the Label class
   - should take an instance of the Item class as an input
   - should add the input item to the collection of items
   - should add self as a property of the item object (by using the correct setter from the item object)
can_be_archived?() in the Book class
   - should override the method from the parent class
   - should return true if parent's method returns true OR if cover_state equals to "bad", otherwise, it should return false
   - Add unit tests for all implemented methods.
- The following options should be available:
   - List all books
   - List all labels (e.g. 'Gift', 'New')
   - Add a book
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created

_@chimwemwe007_ 
- Create MusicAlbum class in a separate .rb file.
- Create Genre class with an association to the Item class (in a separate .rb file).
- All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
   - add_item method in the Genre class
   - should take an instance of the Item class as an input
   - should add the input item to the collection of items
   - should add self as a property of the item object (by using the correct setter from the item object)
   - can_be_archived?() in the MusicAlbum class, should override the method from the parent class
   - should return true if parent's method returns true AND if on_spotify equals true, otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
   - List all music albums
   - List all genres (e.g 'Comedy', 'Thriller')
   - Add a music album
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created

_@bengigo_

- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram should be defined and set up in the constructor method.
- All Author class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
   - add_item method in the Author class
   - should take an instance of the Item class as an input
   - should add the input item to the collection of items
   - should add self as a property of the item object (by using the correct setter from the item object)
   - can_be_archived?() in the Game class should override the method from the parent class
   - should return true if parent's method returns true AND if last_played_at is older than 2 years
otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
   - List of games
   - List all authors (e.g. 'Stephen King')
   - Add a game
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created

catalog_of_my_things

### CODE REVIEW

File: schema/schema.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  FOREIGN KEY(id) REFERENCES genres(id),
+  publish_date DATE,
+  archived BOOLEAN
+);
+
+CREATE TABLE genres(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  name VARCHAR(255)
+  PRIMARY KEY(id)
+);
```
Comment (Nemwel-Boniface) [inline schema/schema.sql:13]: - [ ] I have realized that you did not implement all the project requirements for the database. I can only see two tables and there are supposed to be SQL commands that create the :  1. books  2. music_albums  3. games  4. authors  5. labels  6. genres. These are the total number of tables with all their columns that are supposed to be created. Kindy ensure that you add them all.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @bengigo , @Ney2 and @chimwemwe007    :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your Ruby Capstone project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema/schema.sql (hunk 1 above)
Comment (Ney2) [reply schema/schema.sql:13]: All the database files are implemented in a different schema file. But, we will merge it into one schema file.

**Review by cynthiainga — APPROVED** [review]
Hi @bengigo, @Ney2, and @chimwemwe007, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All linters are passing ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Professional documentation ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**